boards/BMS/Test/Src/Test_Faults.cpp

```diff
@@ -475,4 +579,120 @@ TEST_F(BmsFaultTest, check_precharge_fault)
+        App_CanRx_DCM_Vitals_Heartbeat_Update(true);
+        App_CanRx_PDM_Vitals_Heartbeat_Update(true);
+
+        get_current_ms_fake.return_val += 10;
+        LetTimePass(state_machine, 10);
+        ASSERT_EQ(App_GetInitState(), App_SharedStateMachine_GetCurrentState(state_machine));
+        ASSERT_FALSE(App_CanAlerts_GetFault(BMS_FAULT_MISSING_HEARTBEAT));
+
+        // Fail to check in a heartbeat, BMS should fault
+        get_current_ms_fake.return_val += HEARTBEAT_MONITOR_TIMEOUT_PERIOD_MS - 10U;
+        LetTimePass(state_machine, HEARTBEAT_MONITOR_TIMEOUT_PERIOD_MS - 10U);
+        ASSERT_EQ(App_GetInitState(), App_SharedStateMachine_GetCurrentState(state_machine));
+        ASSERT_FALSE(App_CanAlerts_GetFault(BMS_FAULT_MISSING_HEARTBEAT));
+
+        App_CanRx_FSM_Vitals_Heartbeat_Update(!test_params[i].fsm_mia);
+        App_CanRx_DCM_Vitals_Heartbeat_Update(!test_params[i].dcm_mia);
+        App_CanRx_PDM_Vitals_Heartbeat_Update(!test_params[i].pdm_mia);
+
+        get_current_ms_fake.return_val += 10;
+        LetTimePass(state_machine, 10);
+        ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+        ASSERT_TRUE(App_CanAlerts_GetFault(BMS_FAULT_MISSING_HEARTBEAT));
+
+        // Stay faulted indefinitely 
+        get_current_ms_fake.return_val += 1000;
+        LetTimePass(state_machine, 1000);
+        ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+        ASSERT_TRUE(App_CanAlerts_GetFault(BMS_FAULT_MISSING_HEARTBEAT));
+
+        // Check heartbeats back in, fault should clear and transition back to init
+        App_CanRx_FSM_Vitals_Heartbeat_Update(true);
+        App_CanRx_DCM_Vitals_Heartbeat_Update(true);
+        App_CanRx_PDM_Vitals_Heartbeat_Update(true);
+
+        get_current_ms_fake.return_val += HEARTBEAT_MONITOR_TIMEOUT_PERIOD_MS;
+        LetTimePass(state_machine, HEARTBEAT_MONITOR_TIMEOUT_PERIOD_MS);
+        ASSERT_EQ(App_GetInitState(), App_SharedStateMachine_GetCurrentState(state_machine));
+        ASSERT_FALSE(App_CanAlerts_GetFault(BMS_FAULT_MISSING_HEARTBEAT));
+    }
+}
+
+TEST_F(BmsFaultTest, check_state_transition_to_fault_disables_bms_ok)
+{
+    // Let accumulator startup count expire
+    LetTimePass(state_machine, 1000);
+    ASSERT_EQ(App_GetInitState(), App_SharedStateMachine_GetCurrentState(state_machine));
+    ASSERT_EQ(enable_bms_ok_fake.call_count, 0);
+    ASSERT_EQ(disable_bms_ok_fake.call_count, 0);
+
+    // Set cell voltage critically high and confirm fault is set
+    set_cell_voltage((AccumulatorSegment)0U, 0U, MAX_CELL_VOLTAGE + 0.1f);
+    LetTimePass(state_machine, 20);
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+
+    // BMS OK status should be disabled upon entering fault state, which is latched in hardware
+    ASSERT_EQ(enable_bms_ok_fake.call_count, 0);
+    ASSERT_EQ(disable_bms_ok_fake.call_count, 1);
+}
```

Comment:
Not super rigorous but confirms we at least attempt to latch BMS not OK status upon entering fault state